asynction/security/validation.py

```diff
@@ -0,0 +1,336 @@
+import base64
```

Comment:
This validation module looks very sensible to me 👍 

I think we can reduce the security package to just this module. I'd probably have a top-lvl `security.py` module that will include this validation logic. The rest of the `security` package can be spread across the existing top-lvl modules (as I've explained in the other comments).